tests/integration/test_force_rtl.py

```diff
@@ -0,0 +1,166 @@
+    """
+    Add a takeoff command and waypoint following commands to the drone's
+    command sequence, and upload them.
+
+    Parameters
+    ----------
+    controller: "flight_controller.FlightController"
+    waypoints: "list[tuple[float, float, float]]"
+        The three values in the tuple represent latitude (decimal degrees),
+        longitude (decimal degrees), and altitude (metres) respectively.
+
+    Returns
+    -------
+    bool
+        Returns if the mission is successfully uploaded or not.
+    """
+    # Clear existing mission
+    controller.drone.commands.download()
+    controller.drone.commands.wait_ready()
+    controller.drone.commands.clear()
+
+    # Add takeoff command to the mission
+    takeoff_command = dronekit.Command(
+        0,
+        0,
+        0,
+        MAVLINK_TAKEOFF_FRAME,
+        MAVLINK_TAKEOFF_COMMAND,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        ALTITUDE,
+    )
+
+    controller.drone.commands.add(takeoff_command)
+
+    # Add waypoints to the mission
+    for point in waypoints:
+        command = dronekit.Command(
+            0,
+            0,
+            0,
+            MAVLINK_FRAME,
+            MAVLINK_COMMAND,
+            0,
+            0,
+            0,
+            ACCEPT_RADIUS,
+            0,
+            0,
+            point[0],
+            point[1],
+            point[2],
+        )
+
+        controller.drone.commands.add(command)
+
+    # Upload the mission to the drone
+    try:
+        controller.drone.commands.upload()
+        return True
+    except dronekit.TimeoutError:
+        return False
+
+
+def main() -> int:
+    """
+    Main function.
+    """
+    result, controller = flight_controller.FlightController.create(MISSION_PLANNER_ADDRESS)
+    if not result:
+        print("Failed to create flight controller.")
+        return -1
+
+    # Get Pylance to stop complaining
+    assert controller is not None
+
+    # List of waypoints for the drone to travel
+    waypoints = [
+        (43.4731, -80.5419, ALTITUDE),
+        (43.4723, -80.5380, ALTITUDE),
+        (43.4735, -80.5371, ALTITUDE),
+        (43.4743, -80.5400, ALTITUDE),
+    ]
+
+    # Upload mission
+    result = upload_mission(controller, waypoints)
+    if not result:
+        print("Failed to upload mission.")
+        return -1
+
+    while True:
+        result, is_drone_destination_final_waypoint = (
+            controller.is_drone_destination_final_waypoint()
+        )
+        if not result:
+            print("Failed to get if the drone's destination is the final waypoint.")
+            return -1
+
+        # Get Pylance to stop complaining
+        assert is_drone_destination_final_waypoint is not None
+
+        if is_drone_destination_final_waypoint:
+            break
+
+        print("Drone's destination is not final waypoint.")
+
+        time.sleep(DELAY_TIME)
+
+    print("Drone's destination is final waypoint.")
```

Comment:
I don't think this is needed. You can just have a delay for some time and then call force RTL function.